### DISCUSSION TITLE
When you clear the cache, what happens next?

### DISCUSSION THREAD

**Discussion by Thomas-Richardson** [scambier/obsidian-omnisearch#140] [Q&A]
The plug-in broke for me, and a reinstall didn't work, so I decided to clear the cache. I guess what this does is, makes the plug-in forget about all the notes in my vault? Does it have to rebuild the cache? How long would this take? And how would I know if it was done?

Thanks,

Richie

**Comment by scambier** [comment]
> I guess what this does is, makes the plug-in forget about all the notes in my vault

Basically. It clears the IndexedDb table where the search index is stored. So **you need to reload Obsidian** (the command "Reload app without saving", or a full restart), and Omnisearch will simply rebuild its search index from scratch. It will work the same as the first time you launched the plugin. And indeed, the cache is _not_ cleared when you uninstall the plugin; I will fix that.

> How long would this take? And how would I know if it was done?

In the current beta, Omnisearch will display a notice if there's 500+ files to reindex, but the process shouldn't take more than a few seconds for most vaults. In any case, there will be a message under the search field, indicating that Omnisearch is doing some work. If there is no message, then it's good.
